File: backend/warranty/src/main/java/edu/uth/warranty/model/Invoice.java
package edu.uth.warranty.model;

import jakarta.persistence.*;

import lombok.Getter;
import lombok.Setter;
import lombok.NoArgsConstructor;
import lombok.AllArgsConstructor;

@Entity
@Getter
@Setter
@NoArgsConstructor
@AllArgsConstructor
@Table(name = "Invoice")
public class Invoice {
    @Id
    @GeneratedValue(strategy = GenerationType.IDENTITY)
    private Long invoice_id;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "claim_id", nullable = false)
    private WarrantyClaim claim;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "part_id", nullable = false)
    private Part part;

    @ManyToOne(fetch = FetchType.LAZY)
    @JoinColumn(name = "center_id", nullable = false)
    private ServiceCenter center;

    @Column(name = "location_type", nullable = false)
    private String location_type;

    @Column(name = "quantity", nullable = false)
    private Integer quantity;

    @Column(name = "min_stock_level", nullable = false)
    private Integer min_stock_level;

    @Column(name = "payments_status", nullable = false)
    private String payments_status;
}

File: frontend/src/types/invoice.ts
export interface InvoiceRequest {
    id?: number;
    claimId: number;
    partId: number;
    centerId: number;
    
    locationType: string;
    quantity: number;
    minStockLevel: number;
    paymentsStatus: string;
}

export interface InvoiceResponse {
    id: number;
    claimId: number;
    partId: number;
    partName: string; 
    centerId: number;
    centerName: string;
    
    locationType: string;
    quantity: number;
    minStockLevel: number;
    paymentsStatus: string;
}


